include/oneapi/dpl/internal/random_impl/uniform_real_distribution.h

```diff
@@ -128,210 +128,148 @@ class uniform_real_distribution
     // Distribution parameters
     scalar_type a_;
     scalar_type b_;
+    template <class _Engine>
+    inline scalar_type
+    scale_and_displace(const scalar_type& __in, const param_type& __params, _Engine& __engine)
+    {
+        return ((__in - __engine.min()) / (1 + static_cast<scalar_type>(__engine.max() - __engine.min()))) *
+                   (__params.second - __params.first) +
+               __params.first;
+    }
 
     // Implementation for generate function
-    template <int _Ndistr, int _Negnine, class _Engine>
-    typename ::std::enable_if<((_Ndistr == _Negnine) & (_Ndistr != 0)), result_type>::type
+    template <int _Ndistr, int _Nengine, class _Engine>
+    typename ::std::enable_if<((_Ndistr == _Nengine) & (_Ndistr != 0)), result_type>::type
     generate(_Engine& __engine, const param_type& __params)
     {
         auto __engine_output = __engine();
         result_type __res;
 
         for (int __i = 0; __i < _Ndistr; ++__i)
-            __res[__i] = static_cast<scalar_type>(__engine_output[__i]);
-
-        __res = ((__res - __engine.min()) / (1 + static_cast<scalar_type>(__engine.max() - __engine.min()))) *
-                    (__params.second - __params.first) +
-                __params.first;
+            __res[__i] = scale_and_displace(static_cast<scalar_type>(__engine_output[__i]), __params, __engine);
```

Comment:
How does it affect performance on different devices?

Reply:
It would be better to keep vec version of scale_and_displace function here
static_cast + loop was used instead of convert() function as convert() has a lower performance - this issue probably already fixed

Reply:
changed to vectorized approach

Reply:
will be moved to a single commit as it causes problems on windows which are needed to be investigated